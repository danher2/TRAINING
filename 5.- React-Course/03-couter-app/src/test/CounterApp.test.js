import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import React from "react";
import CounterApp from "../CounterApp";
import { shallow } from 'enzyme';

describe('debe mostrar <CounterApp/> correctamente (hacer match con snapshot) y sus valores por defecto', ()=>{

    let wrapper = shallow(<CounterApp />);

    beforeEach(()=>{
        console.log('se ejecuta antes de cada test')
         wrapper = shallow(<CounterApp />);


    })


    test('mostrar <CounterApp/> correctamente (hacer match con snapshot)',()=>{
            expect(wrapper).toMatchSnapshot();

    })


    test(' mostrar el valor por defecto de 100',()=>{

        const wrapper = shallow(<CounterApp value = {100} />);
        const wrapperCounter = wrapper.find('h2').text().trim();
        console.log(wrapperCounter);
        expect(wrapperCounter).toBe('100')

})

    test('debe de incrementar con el boton de +1 ', () => {
     const bt1 =   wrapper.find('button').at(0).simulate('click');
        // console.log(bt1.html());

        const wrapperCounter = wrapper.find('h2').text().trim();
        console.log(wrapperCounter);
       
        expect(wrapperCounter).toBe('11');

    })



    test('debe de decrementar con el boton de -1 ', () => {

        
        const bt1 =   wrapper.find('button').at(2).simulate('click');
           // console.log(bt1.html());
   
           const wrapperCounter = wrapper.find('h2').text().trim();
           console.log(wrapperCounter);
          
           expect(wrapperCounter).toBe('9');
   
       })
       

       test('debe de colocar el valor por defecto con el boton btn reset', () => {

        const wrapper = shallow(<CounterApp value = {105} />);

        //simulando click en los botones
        wrapper.find('button').at(0).simulate('click'); //boton sumar
        wrapper.find('button').at(0).simulate('click');// boton sumar
        wrapper.find('button').at(1).simulate('click');// boton reset 
        const wrapperCounter = wrapper.find('h2').text().trim();
        console.log(wrapperCounter);   

        expect(wrapperCounter).toBe('105')
   
       })


})