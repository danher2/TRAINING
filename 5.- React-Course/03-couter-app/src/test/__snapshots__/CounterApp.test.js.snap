// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe mostrar <CounterApp/> correctamente (hacer match con snapshot) y sus valores por defecto mostrar <CounterApp/> correctamente (hacer match con snapshot) 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
